app.py

```diff
@@ -1,74 +1,127 @@
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#issues-events """
+    try:
+        a = body["object_attributes"]["action"]
+    except KeyError:
+        a = "Touch"
     if a == "update":
         return False
     body.update({"action": (a + "d") if a[-1] == "e" else (a + "ed")})
     return """:page_facing_up: <{project[web_url]}/issues/{object_attributes[id]}>
 **{user[name]}** {action} an issue: {object_attributes[title]}""".format(**body)
 
+
 def handle_note(body):
     """ Handle GitLab comment (note) event webhook
-    https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#comment-events """
+    https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#comment-events """
+
     def convert(type):
         s1 = sub('(.)([A-Z][a-z]+)', r'\1 \2', type)
         return sub('([a-z0-9])([A-Z])', r'\1 \2', s1).lower()
-    body.update({"type": convert(body["object_attributes"]["noteable_type"]) })
+
+    body.update({"type": convert(body["object_attributes"]["noteable_type"])})
     return """:notepad_spiral: <{object_attributes[url]}>
 **{user[name]}** commented on a {type}:
 {object_attributes[note]}""".format(**body)
 
+
 def handle_merge(body):
     """ Handle GitLab merge request event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#merge-request-events """
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#merge-request-events """
+    try:
+        url = body["object_attributes"]["url"]
+    except KeyError:
+        url = body["project"]["web_url"] + "/merge_requests"
+    body["object_attributes"]["url"] = url
     return """:arrows_counterclockwise: <{object_attributes[url]}>
-**{user[name]}** created a merge request: {object_attributes[source_branch]}->{object_attributes[target_branch]} **{object_attributes[title]}**""".format(**body)
+**{user[name]}** created a merge request: {object_attributes[source_branch]}->{object_attributes[target_branch]} **{object_attributes[title]}**""".format(
+        **body)
+
 
 def handle_wiki(body):
     """ Handle GitLab wiki page event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#wiki-page-events """
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#wiki-page-events """
     return """:notebook: <{object_attributes[url]}>
 **{user[name]}** created a wiki page: {object_attributes[title]}""".format(**body)
 
+
 def handle_pipeline(body):
     """ Handle GitLab pipeline event webhook
-        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md#pipeline-events """
+        https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/user/project/integrations/webhooks.md#pipeline-events """
     # i don't even know what to put here
```

Comment:
Thank you for implementing pipelines for me, I have no need for them myself so I'm happy to leave these functions as they are. Just remove this comment line for me please, it's not needed anymore.

Reply:
Hehe missed that one :)